src/vmspawn/vmspawn.c

```diff
@@ -370,11 +581,48 @@ static int run_virtual_machine(void) {
                 r = strv_extendf(&cmdline, "type=11,value=io.systemd.stub.kernel-cmdline-extra=%s", kcl);
                 if (r < 0)
                         return log_oom();
+                log_debug("kcl=%s", kcl);
 #else
                 log_warning("Cannot append extra args to kernel cmdline, native architecture doesn't support SMBIOS");
 #endif
         }
 
+#ifdef ARCHITECTURE_SUPPORTS_SMBIOS
+        if (use_vsock) {
+                r = open_vsock();
+                if (r < 0)
+                        return log_error_errno(r, "Failed to open vsock: %m");
+                vsock_fd = r;
+
+                r = strv_extend(&cmdline, "-smbios");
+                if (r < 0)
+                        return log_oom();
+
+                union sockaddr_union addr;
+                socklen_t addr_len = sizeof addr.vm;
+                r = getsockname(vsock_fd, &addr.sa, &addr_len);
+                if (r < 0)
+                        return log_error_errno(errno, "Failed to call getsockname on vsock: %m");
+                if (addr_len < sizeof addr.vm)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "getsockname didn't return enough data: %m");
+                if (addr.vm.svm_family != AF_VSOCK)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "getsockname returned data for the wrong socket type: %m");
```

Comment:
both of these checks could probably be assert()s, as this would be an api violation of the kernel

Reply:
Fixed in 7126b0eb12.